packages/fxa-content-server/app/scripts/views/mixins/email-autocomplete-domains-mixin.js

```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A mixin to autocomplete common domains from `datalist option` `value`s.
+ */
+import Template from 'templates/partial/email-autocomplete-domains.mustache';
+
+export const EMAIL_SELECTOR = 'input[type=email]';
+export const DATALIST_OPTIONS_SELECTOR = '#autocomplete-domain option';
+export const FOCUS_HACK_SELECTOR = '#focus-hack';
+
+export const DOMAINS = [
+  'gmail.com',
+  'outlook.com',
+  'hotmail.com',
+  'yahoo.com',
+  'qq.com',
+  'web.de',
+  'aol.com',
+  'mail.ru',
+  'icloud.com',
+  'gmx.de',
+  't-online.de',
+  'orange.fr',
+  'yandex.ru',
+  'yahoo.fr',
+  'live.com',
+  '163.com',
+  'msn.com',
+  'comcast.net',
+  'hotmail.co.uk',
+  'hotmail.fr',
+];
```

Comment:
@chenba I know you mentioned in #3197 we should stay synced with what's in `mailcheck.js` but this list seems a little more distinct than what we've got in there. I could move these into that file. 🤷‍♀️

Also, @davismtl, you may want to verify these are what you're wanting.

Reply:
That list seems accurate. Same 20 I had in mind.

We can look into expanding it in the future but it's an excellent start.

Reply:
👍 The order shown here is also going to be the order the list pulls up in the dropdown for desktop & mobile so I checked that Amplitude link you sent before and went in that order. I know I showed this in the demo yesterday but top 3 are shown in the keyboard in mobile if the user types and then backspaces to the "@" sign.

<img width="426" alt="image" src="https://user-images.githubusercontent.com/13018240/69285046-a09f4500-0bb5-11ea-93a8-be344ea9529d.png">